tests/org.jboss.ide.eclipse.as.ui.bot.test/src/org/jboss/ide/eclipse/as/ui/bot/test/template/DeployJSPProjectTemplate.java

```diff
@@ -91,8 +91,17 @@ public void clearServerConsole(){
 	
 	@Test
 	public void deployProject(){
-		log.step("Assert project is build");
-		assertProjectIsBuilt();
+		log.step("Assert project is built");
+		try {
+			assertProjectIsBuilt();
+		} catch (Exception e){
+			log.step("Refresh project");
+			getProject().refresh();
+			log.step("Re-assert project is built");
+			assertProjectIsBuilt();
+			log.step("Re-throw exception");
```

Comment:
Are you sure you want to rethrow exception? It means test will failed and not continue after test project is refreshed